File: Microsoft Office Developer Documentation Team/Outlook-2010-Outlook-Social-Connector-OSC-3977/[C#]-Outlook 2010 Outlook Social Connector (OSC) Provider Templates/C#/OSCProvider_CS/OSCProfile.cs
using System;
using System.Diagnostics;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using OutlookSocialProvider;

namespace OSCProvider_CS
{
    class OSCProfile : OSCPerson, OutlookSocialProvider.ISocialProfile 
    {
        #region ISocialProfile Members

        public bool[] AreFriendsOrColleagues(string[] userIDs)
        {
            //Not supported in OSC version 1.0 and version 1.1
            return null;
        }

        public string GetActivitiesOfFriendsAndColleagues(DateTime startTime)
        {
            //To-do: Implement GetActivitiesOfFriendsAndColleagues
            return string.Empty;
        }

        public void SetStatus(string Status)
        {
            //Not supported in OSC version 1.0 and version 1.1
        }

        #endregion

        #region ISocialPerson members
        public new string GetActivities(DateTime startTime)
        {
            return string.Empty;
        }


        public new string GetDetails()
        {
            //To-do: Implement GetDetails
            return string.Empty;
        }

        public new string GetFriendsAndColleagues()
        {
            //To-do: Implement GetFriendsAndColleagues
            return string.Empty;
        }

        public new string[] GetFriendsAndColleaguesIDs()
        {
            //To-do: Implement GetFriendsAndColleaguesIDs
            return null;
        }

        public new byte[] GetPicture()
        {
            //To-do: Implement GetPicture
            return null;
        }

        public new string GetProfileUrl()
        {
            //To-do: Implement GetDetails
            return string.Empty;
        }

        public new string GetStatus()
        {
            //Not supported in OSC version 1.0 and version 1.1
            throw new NotImplementedException();
        }


        #endregion
    }
}

File: Microsoft Office Developer Documentation Team/Outlook-Social-Connector-2013-Provider-templates-17387/[VB]-Outlook Social Connector 2013 Provider templates/VB/OSCProvider_VB/OSCProfile.vb
Imports System
Imports System.Data
Imports System.Xml
Imports System.Xml.Serialization
Imports System.Diagnostics
Imports System.Drawing
Imports System.Text
Imports OutlookSocialProvider

Namespace OSCProvider_VB
    Public Class OSCProfile
        Implements OutlookSocialProvider.ISocialProfile

#Region "ISocialProfile Members"
        Public Function GetActivities(ByVal startTime As Date) As String _
            Implements OutlookSocialProvider.ISocialPerson.GetActivities, _
            OutlookSocialProvider.ISocialProfile.GetActivities
            'Not supported since OSC 2013
            Return String.Empty
        End Function

        Public Function GetDetails() As String _
            Implements OutlookSocialProvider.ISocialPerson.GetDetails, _
            OutlookSocialProvider.ISocialProfile.GetDetails
            'To-Do: Implement GetDetails
            Return String.Empty
        End Function

        Public Function GetFriendsAndColleagues() As String _
            Implements OutlookSocialProvider.ISocialPerson.GetFriendsAndColleagues, _
            ISocialProfile.GetFriendsAndColleagues
            'To-Do: Implement GetFriendsAndColleagues
            Return String.Empty
        End Function

        Public Function GetFriendsAndColleaguesIDs() As String() _
            Implements OutlookSocialProvider.ISocialPerson.GetFriendsAndColleaguesIDs, _
            OutlookSocialProvider.ISocialProfile.GetFriendsAndColleaguesIDs
            'To-Do: Implement GetFriendsAndColleaguesIDs
            Return Nothing
        End Function

        Public Function GetPicture() As Byte() _
            Implements OutlookSocialProvider.ISocialPerson.GetPicture, _
            OutlookSocialProvider.ISocialProfile.GetPicture
            'To-Do: Implement GetPicture
            Return Nothing
        End Function

        Public Function GetStatus() As String _
            Implements OutlookSocialProvider.ISocialPerson.GetStatus, _
            OutlookSocialProvider.ISocialProfile.GetStatus
            'Not supported
            Return String.Empty
        End Function

        Public Function AreFriendsOrColleagues(ByVal userIDs As String()) As Boolean() _
            Implements OutlookSocialProvider.ISocialProfile.AreFriendsOrColleagues
            'Not supported in OSC version 1.0 and version 1.1
            Dim result As Boolean() = {True}
            Return result
        End Function

        Public Function GetActivitiesOfFriendsAndColleagues(ByVal startTime As Date) As String _
            Implements OutlookSocialProvider.ISocialProfile.GetActivitiesOfFriendsAndColleagues
            'Not supported since OSC 2013
            Return String.Empty
        End Function

        Public Sub SetStatus(ByVal status As String) _
        Implements OutlookSocialProvider.ISocialProfile.SetStatus
            'Not supported
        End Sub

#End Region
    End Class
End Namespace

